tcms/core/tests/test_middleware.py

```diff
@@ -0,0 +1,23 @@
+import os
+import unittest
+from django.test import TransactionTestCase
+from django.contrib.messages import get_messages
+
+
+@unittest.skipUnless(
+    os.getenv('TEST_CHECK_UNAPPLIED_MIGRATIONS_MIDDLEWARE'),
+    'CheckUnappliedMigrationsMiddleware testing not enabled')
+class TestCheckUnappliedMigrationsMiddleware(TransactionTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.unapplied_migration_message = ('You have 41 unapplied migration(s)'
```

Comment:
the number will change both when we add our own migrations and when Django or any of the other dependencies does, which will break this test.  Semi-auto upgrades like the ones with PyUP bot and Dependabot will become more fragile and require manual adjusting of this value.

Instead try asserting on parts of the message + the URL, it should be enough.